workshop1/slides/src/slides.md

```diff
@@ -0,0 +1,664 @@
+---
+title: "Haskell Workshop"
+subtitle: "Writing your first Haskell programm"
+author: "Nicolas Audinet"
+institute: "RELEX Solutions Dev Day"
+date: 2019-04-12
+#background-image: src/solarized-mountains.png
+#background-image: vector/relex-logo-black-rgb.pdf
+titlegraphic: vector/relex-logo-black-rgb.pdf
+classoption: "aspectratio=169"
+classoption: "13pt"
+outertheme: "infolines"
+innertheme: "circles"
+colortheme: "dove"
+navigation: "frame"
+lang: "en-US"
+---
+
+## The Goal
+
+Develop a complete Haskell program that converts CSV data to JSON.
+
+![](images/goal.pdf)
+
+## The Data
+
+![](images/data.png){ height=90% }
+
+## The Plan
+
+1) Haskell Development Environment
+2) Hello World and some Syntax
+3) The Data Model
+4) Data Transformation
+5) Connecting to the Outside
+6) Conclusion
+
+# The Haskell Development Environment
+
+## Setting up your environment
+
+- Go to [the Github repo](https://github.com/relex/haskell-workshop) for the workshop
+(https://github.com/relex/haskell-workshop)
+
+- Clone it locally
+
+- Go to workshop1/Exercise.md
+
+- Complete "Set up tooling" section
+
+## Stack
+
+* One of Haskell's build tools.
+* To build a project: `stack build`
+* To access the REPL: `stack repl`
+
+![](images/basic-repl.png)
+
+# Hello Syntax
+
+## Playing with the REPL
+
+* Useful tool to try out parts of your program
+* Extensively used in the workshop
+
+<!-- Live REPL Coding
+* Let binding
+* Numbers & mathematical operators
+* Strings
+* Defining a function
+-->
+
+## Hello World
+
+* Open src/Main.hs
+* Run in the REPL by typing `:main`
+
+```haskell
+main :: IO ()
+main = putStrLn "Hello World!"
+```
+
+* `main` is special:
+  - Present in every Haskell program
+  - No arguments
+  - Performs an IO action
+
+* Change the text and run it in the REPL
+
+## Functions and Types
+
+* The primary way of defining computation is with **functions**
+* **Types** describe the inputs and outputs of functions
+* Types are **enforced** by the compiler
+* Working with two languages at once :)
+
+```haskell
+identity :: Int -> Int
+identity x = x
+
+hello :: String -> String
+hello name = "Hello, " ++ name
+
+addInt :: Int -> Int -> Int
+addInt x y = x + y
+```
+
+Add these ^ to a file and try them out!
+
+## A First Function
+
+* A function that converts (some) integers to words
+* Pattern matching
+
+```haskell
+intToWord :: Int -> String
+intToWord 1 = "one"
+intToWord 2 = "two"
+intToWord 3 = "three"
+intToWord _ = "dunno"
+```
+
+* Try and write `wordToInt :: String -> Int`
+
+## Polymorphic Types
+
+* Some functions have the same behaviour for values of different types
+  * E.g. the `identity` function
+* Can generalise functions by using *type variables*
+
+```haskell
+identity :: a -> a
+identity x = x
+```
+
+* Can also *constrain* the type variables
+
+```haskell
+add :: Num a => a -> a -> a
+add x y = x + y
+```
+
+* These constraints can be defined using *type classes*
```

Comment:
Add a note for presenter that we will mention them in passing later?

Reply:
Done